mycpp/gc_str.cc

```diff
@@ -286,6 +286,11 @@ BigStr* BigStr::rjust(int width, BigStr* fillchar) {
 }
 
 BigStr* BigStr::replace(BigStr* old, BigStr* new_str) {
+  // Use -1 as in python2: "aaaa".replace(-1) -> "AAAA"
```

Comment:
Oh interesting, now I see that Python 2 replace() is inconsistent with Python re.sub !!   funny

0 means 0 occurences, so yeah that's a good justification

It also silently accepts any negative number, so I think that's fine for us as well.  I think we just make sure that -2 and -99 behave exactly the same way as -1.